optuna/samplers/tpe/sampler.py

```diff
@@ -390,7 +388,7 @@ def _gmm_log_pdf(
         return return_val
 
     def _sample_from_categorical_dist(self, probabilities, size=()):
-        # type: (Union[np.ndarray, np.ndarray], Tuple) -> Union[np.ndarray, np.ndarray]
+        # type: (np.ndarray, Tuple) -> np.ndarray
```

Comment:
This function is called at following line only.

https://github.com/pfnet/optuna/blob/fa591a79468a5470191478f4a17d5e8e9da63563/optuna/samplers/tpe/sampler.py#L276-L279

`weighted_below` is 1d ndarray of floats (`counts_below` is 1d array of ints, see [numpy.bincount](https://docs.scipy.org/doc/numpy/reference/generated/numpy.bincount.html). `self.prior_weight` is a float object.)

Reply:
And I guess following if statements doesn't effects anything because size is a tuple object which contains one float object.
https://github.com/pfnet/optuna/blob/fa591a79468a5470191478f4a17d5e8e9da63563/optuna/samplers/tpe/sampler.py#L399-L404

Reply:
>  I guess we can refactor the type hints of this function more 

I agree with you.

Reply:
I picked 67cc602 in this branch 👍